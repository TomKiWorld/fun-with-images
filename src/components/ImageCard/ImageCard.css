.ImageCard {
  width: 22rem;
  height: 17rem;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  margin: 1rem;
  position: relative;
}

.ImageCard .cta-button {
  width: 10rem;
  position: absolute;
  bottom: 10%;
  left: 6rem;
}
