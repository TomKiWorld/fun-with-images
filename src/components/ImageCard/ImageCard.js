import React from 'react';
import './ImageCard.css';

/**
 * Image card to display submitted images with option to resubmit
 * 
 * Required props:
 * - id => Image id for css selector
 * - url => Image url to display as background
 * - onResubmit => Function to resubmit the image 
 */
const ImageCard = ({id, url, onResubmit}) => {
  return (
    <div id={`ImageCard-${id}`} className='ImageCard' style={{backgroundImage:`url("${url}")`}}>
      <button className='card-button' onClick={() => onResubmit(url)}>Resubmit</button>
    </div>
  );
}

export default ImageCard;
