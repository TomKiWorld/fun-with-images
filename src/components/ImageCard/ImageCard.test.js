import { shallow } from 'enzyme';
import React from 'react';
import ImageCard from './ImageCard';

describe('ImageCard Component', () => {
  const mockProps= {
    resubmit: false
  };
  const onMockResubmit = () => {
    mockProps.resubmit= true;
  };
  const wrapper= shallow(
    <ImageCard 
      id={1} 
      url={'https://i.mdel.net/mdx/i/2019/01/2019_01_16_Models.com_15.jpg'}
      onResubmit={onMockResubmit}
      />     
  );
  const resubmitBtn = wrapper.find('.card-button').at(0);

  it('Expect to render ImageCard Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Expect to resubmit the image', () => {
    resubmitBtn.simulate('click')
    expect(mockProps.resubmit).toBe(true);
  });
});