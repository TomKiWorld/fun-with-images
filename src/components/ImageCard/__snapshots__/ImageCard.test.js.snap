// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard Component Expect to render ImageCard Component 1`] = `ShallowWrapper {}`;
